lib/toSyllabicSymbols.js

```diff
@@ -0,0 +1,144 @@
+const toUnicode = require( './toUnicode' )
+const { getRegexGroup } = require( './regex-utils' )
+
+// Some symbols represent multiple characters in a single unicode entity
+// These should be considered unsplittable and unmodifiable by other vowels/signs
+// Therefore can calculate their syllabic weight as is
+const oneUnitSyllables = [ 'ਇ', 'ਉ' ]
+const twoUnitSyllables = [ 'ਊ', 'ਓ', 'ਈ', 'ਏ', 'ਐ', 'ਆ', 'ਔ', 'ਆਂ', 'ਨੂੰ' ]
+
+// Other symbols/signs have no weight, therfore remove before processing
+const zeroUnitCharacters = [
+  '੍ਰ',
+  'ੵ',
+  'ੑ',
+  '੍ਵ',
+  '੍ਹ',
+  '੍ਟ',
+  '੍ਨ',
+  '੍ਯ',
+  '੍ਚ',
+  '੍ਤ',
+  '੍',
+  'ਿ',
+  'ੁ',
+  'ਂ',
+  'ਃ',
+  '।',
+  '॥',
+  '☬',
+  '਼',
+  '❁',
+]
+
+// For the rest, need to analyze the string
+// Any base character on it's own counts as one syllable
+// Adding any number of long/deeragh sounds will count as two syllables
+const syllableIdentifier = {
+  baseCharacter: 'b',
+  deeraghModifier: 'a',
+}
+
+// The list of base characters for analysis
+const baseCharacters = [
+  'ਙ',
+  'ੳ',
+  'ਅ',
+  'ਬ',
+  'ਭ',
+  'ਚ',
+  'ਛ',
+  'ਦ',
+  'ਧ',
+  'ੲ',
+  'ਡ',
+  'ਢ',
+  'ਗ',
+  'ਘ',
+  'ਹ',
+  'ਜ',
+  'ਝ',
+  'ਕ',
+  'ਖ',
+  'ਲ',
+  'ਲ਼',
+  'ਮ',
+  'ਨ',
+  'ਪ',
+  'ਫ',
+  'ਤ',
+  'ਥ',
+  'ਰ',
+  'ਸ',
+  'ਸ਼',
+  'ਟ',
+  'ਠ',
+  'ਵ',
+  'ੜ',
+  'ਣ',
+  'ਯ',
+  'ਜ਼',
+  'ਗ਼',
+  'ਖ਼',
+  'ਫ਼',
+  'ਞ',
+]
+
+// The list of long sound characters for analysis
+const deeraghModifiers = [
+  'ੰ',
+  '੍ਹੂ',
+  'ੀ',
+  'ੂ',
+  'ੇ',
+  'ੈ',
+  'ੋ',
+  'ੌ',
+  'ਾਂ',
+  'ਾ',
+  'ੱ',
+]
+
+// Map one-unit syllables to the string '1' and two-unit syllables to the string '2'
+// Otherwise map each character to a syllableIdentifier, for further processing/analysis
+const syllabicMapping = Object.entries( {
+  1: [ ...oneUnitSyllables ],
+  2: [ ...twoUnitSyllables ],
+  [ syllableIdentifier.baseCharacter ]: [ ...baseCharacters ],
+  [ syllableIdentifier.deeraghModifier ]: [ ...deeraghModifiers ],
+} ).reduce( ( newObject, [ weight, characters ] ) => {
+  const characterMap = characters.reduce( ( characterMapObject, [ character ] ) => ( {
+    ...characterMapObject,
+    [ character ]: weight,
+  } ), {} )
+  return ( {
+    ...newObject,
+    ...characterMap,
+  } )
+}, {} )
+
+// Add any missing rules which do not fit in the above patterns
+syllabicMapping[ 'ੴ' ] = '21 2221' // Ik Oankaar / ਇੱਕ ਓਅੰਕਾਰ
```

Comment:
Instead of mutating it, do the assignment directly in the initial parameter of the reducer, or use Object expansion:

Object expansion example:
```
const syllabicMapping = {
  ੴ: '21 2221',
  ' ': ' ',
  ...Object.entries( {
    1: oneUnitSyllables,
    2: twoUnitSyllables,
    [ syllableIdentifier.baseCharacter ]: baseCharacters,
    [ syllableIdentifier.deeraghModifier ]: deeraghModifiers,
  } ).reduce( ( characterSymbols, [ weight, characterSet ] ) => ( {
      ...characterSymbols,
      ...characterSet.reduce( ( characters, [ character ] ) => ( {
      ...characters,
      [ character ]: weight,
    } ), {} ),
  }, {} )
```

Reducer initial parameter example:
```
const syllabicMapping = Object.entries( {
  1: oneUnitSyllables,
  2: twoUnitSyllables,
  [ syllableIdentifier.baseCharacter ]: baseCharacters,
  [ syllableIdentifier.deeraghModifier ]: deeraghModifiers,
} ).reduce( ( characterSymbols, [ weight, characterSet ] ) => ( {
    ...characterSymbols,
    ...characterSet.reduce( ( characters, [ character ] ) => ( {
    ...characters,
    [ character ]: weight,
  } ), {} ),
}, {
  ੴ: '21 2221',
  ' ': ' ',
} )
```

I tend to prefer Object Expansion because it's immediately obvious that these are syllabic mapping symbols, whereas in a reducer, it could be an initial value that is transformed (and you have to read the code to find out that it's simply appended)

Reply:
If it looks messy to inline the character sets inside `Object.entries`, move the object into a separate `characterSets` object above